ddev/ddev
# Fix gitpod failure to start, fixes #4889

## The Issue

* #4889 

Due to apparent odd docker configuration, DDEV projects fail to start on gitpod

## How This PR Solves The Issue

Instead of using the classic approach of writing to /proc/1/fd/1 to get info into docker logs, use a dedicated fifo.

## TODO

- [x] Make sure we fail quickly and with good information if start.sh fails. Maybe we should be running `bash -c "start.sh || kill 1"`, something like that. 
- [x] Make sure that when container is started NOT in ddev context that the logs are done correctly. If pipe doesn't exist, create link, start a cat in the background.


## Manual Testing Instructions

Start a project on gitpod. 
Make sure it works
Use `ddev logs` to make sure expected info is there.

## Automated Testing Overview

No changes

## Related Issue Link(s)

## Release/Deployment Notes





<a href="https://gitpod.io/#https://github.com/ddev/ddev/pull/4895"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
